Dapper.Contrib/SqlMapperExtensions.cs

```diff
@@ -628,27 +659,41 @@ public class ComputedAttribute : Attribute
 public partial interface ISqlAdapter
 {
     int Insert(IDbConnection connection, IDbTransaction transaction, int? commandTimeout, String tableName, string columnList, string parameterList, IEnumerable<PropertyInfo> keyProperties, object entityToInsert);
+    
+    //new methods for issue #336
+    void AppendColumnName(StringBuilder sb, string columnName);
+    void AppendColumnNameEqualsValue(StringBuilder sb, string columnName);
 }
 
 public partial class SqlServerAdapter : ISqlAdapter
 {
     public int Insert(IDbConnection connection, IDbTransaction transaction, int? commandTimeout, String tableName, string columnList, string parameterList, IEnumerable<PropertyInfo> keyProperties, object entityToInsert)
     {
-
         var cmd = String.Format("insert into {0} ({1}) values ({2});select SCOPE_IDENTITY() id", tableName, columnList, parameterList);
-        var multi = connection.QueryMultiple(cmd,entityToInsert , transaction, commandTimeout);
+        var multi = connection.QueryMultiple(cmd, entityToInsert, transaction, commandTimeout);
 
-        var id = (int)multi.Read().First().id;
-        var propertyInfos = keyProperties as PropertyInfo[] ?? keyProperties.ToArray();
-        if (!propertyInfos.Any()) return id;
+        var first = multi.Read().FirstOrDefault();
+        if (first == null || first.id == null) return 0;
+
+        var id = (int)first.id;
+        var pi = keyProperties as PropertyInfo[] ?? keyProperties.ToArray();

```

Comment:
True, I was not sure about changing the variable names actually and I'm thinking of going back to the old, more readable names!